packages/inngest/src/components/connect/index.ts

```diff
@@ -129,12 +138,27 @@ class WebSocketWorkerConnection implements WorkerConnection {
     });
   }
 
-  private get functions(): InngestFunction.Any[] {
-    return (
-      (this.options.functions as InngestFunction.Any[]) ??
-      this.inngest["localFns"] ??
-      []
-    );
+  private get functions(): Record<
+    string,
+    {
+      client: Inngest.Like;
+      functions: InngestFunction.Any[];
+    }
+  > {
+    const functions: Record<
+      string,
+      {
+        client: Inngest.Like;
+        functions: InngestFunction.Any[];
+      }
+    > = {};
+    for (const app of this.options.apps) {
+      functions[app.client.id] = {
+        client: app.client,
+        functions: (app.functions as InngestFunction.Any[]) ?? [],
+      };
+    }
```

Comment:
We could find that some app IDs collide here; maybe we should log a warning if this happens.